### ISSUE TITLE
Error after upgrading from 2.2 to 3.0. Routevaluedictinoary could not be found

### ISSUE THREAD

**Issue by esmeraldahs** [aspnet/AspNetCore#16895]
Hello,
I have upgraded my solution from .net core 2.2 to 3.0 following the guide here https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.0&tabs=visual-studio#framework-reference

After upgrading i  am getting this error: 
Reference to type 'RouteValueDictionary' claims it is defined in 'Microsoft.AspNetCore.Routing.Abstractions', but it could not be found.
I already read the issue [here](https://github.com/aspnet/AspNetCore/issues/15261) but none of the solutions/suggstions seemed to do the work and i am not using Telerik package in my solution. 

The code where the error happens: 
![image](https://user-images.githubusercontent.com/19766388/68405250-089a6800-0180-11ea-9263-d5091d275585.png)

I am using ReflectionIT.Mvc.Paging for pagination and filtering, and i suppose maybe it is not supported in .net core 3.0 anymore..

Contents of csproj file: 

`<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <UserSecretsId>userSecretId</UserSecretsId>
    <ApplicationIcon>wwwroot\favicon.ico</ApplicationIcon>
    <LangVersion>latest</LangVersion>
  </PropertyGroup>
  <ItemGroup>
    <FrameworkReference Include="Microsoft.AspNetCore.App" />
    <PackageReference Include="Audit.NET" Version="14.7.0" />
    <PackageReference Include="Audit.NET.SqlServer" Version="13.1.4" />
    <PackageReference Include="BeginCollectionItemCore" Version="1.0.8" />
    <PackageReference Include="Microsoft.ApplicationInsights" Version="2.11.0" />
    <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.8.2" />
    <PackageReference Include="Microsoft.AspNet.WebApi.Client" Version="5.2.7" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.OpenIdConnect" Version="3.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.0.0" />
    <PackageReference Include="Microsoft.Azure.KeyVault" Version="3.0.4" />
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="2.2.0" />
    <PackageReference Include="Microsoft.IdentityModel.Clients.ActiveDirectory" Version="5.2.4" />
    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="2.1.9" />
    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
    <PackageReference Include="ReflectionIT.Mvc.Paging" Version="3.5.0" />
    <PackageReference Include="Serilog" Version="2.9.0" />
    <PackageReference Include="Serilog.AspNetCore" Version="3.1.0" />
    <PackageReference Include="Serilog.Sinks.ApplicationInsights" Version="3.0.4" />
    <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
    <PackageReference Include="Serilog.Sinks.File" Version="4.1.0" />
    <PackageReference Include="Serilog.Sinks.RollingFile" Version="3.3.0" />
    <PackageReference Include="Serilog.Sinks.RollingFileAlternate" Version="2.0.9" />
    <PackageReference Include="System.Runtime.Caching" Version="4.5.0" />
    <PackageReference Include="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Version="3.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="3.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Identity.UI" Version="3.0.0" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.0.0" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="3.0.0" />
  </ItemGroup>
  <ItemGroup>
    <Content Update="wwwroot\favicon.ico">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
  </ItemGroup>
  <ItemGroup>
    <Folder Include="wwwroot\font\" />
  </ItemGroup>
  <ItemGroup>
    <WCFMetadata Include="Connected Services" />
  </ItemGroup>
</Project>
`

Any help is really appreciated!

**Comment by javiercn** [comment]
@esmeraldahs thanks for contacting us.

Can you try and remove 3rd party packages and see if that fixes the issue? It is likely a 3rd party package that needs to be upgraded.

**Comment by esmeraldahs** [comment]
Sorry but in this scenario which is considered a 3rd party package?

**Comment by javiercn** [comment]
@esmeraldahs Something that is not part of ASP.NET Core

**Comment by esmeraldahs** [comment]
Ah okay, thank you. Will do and let you know.

**Comment by esmeraldahs** [comment]
If i remove those packages, i would have to comment 50% of my code.
I am thinking that probably  ReflectionIT.Mvc.Paging is not compatible with .net core 3.0

**Comment by javiercn** [comment]
@esmeraldahs 

I've tracked this down. Seems like it was missing a typeforward during the RTM time-frame, but that it was added recently. https://github.com/aspnet/AspNetCore/blob/master/src/Http/Routing.Abstractions/ref/Microsoft.AspNetCore.Routing.Abstractions.Manual.cs

I believe this will be available in the next patch release. If you update to the latest 3.1 preview that will likely fix the issue too and unblock you.
